Maskbook/Extension/UIView+Shake.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  UIView+Shake.swift
+//  Maskbook
+//
+//  Created by xiaojian sun on 2022/2/23.
+//  Copyright © 2022 dimension. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+enum ShakeDirection {
+    case horizontal
+    case vertical
+}
+// swiftlint:disable function_parameter_count legacy_random multiline_arguments
+extension UIView {
+    private func degreesToRadians(_ x: CGFloat) -> CGFloat {
+        .pi * x / 180.0
+    }
+
+    func shakeWhenDelete(duration: Double = 0.25,
+                         displacement: CGFloat = 1.0,
+                         degreesRotation: CGFloat = 2.0)
+    {
+        let negativeDisplacement = -1.0 * displacement
+        let position = CAKeyframeAnimation(keyPath: "position")
+        position.beginTime = 0.8
+        position.duration = duration
+        position.values = [
+            NSValue(cgPoint: CGPoint(x: negativeDisplacement, y: negativeDisplacement)),
+            NSValue(cgPoint: CGPoint(x: 0, y: 0)),
+            NSValue(cgPoint: CGPoint(x: negativeDisplacement, y: 0)),
+            NSValue(cgPoint: CGPoint(x: 0, y: negativeDisplacement)),
+            NSValue(cgPoint: CGPoint(x: negativeDisplacement, y: negativeDisplacement))
+        ]
+        position.calculationMode = CAAnimationCalculationMode.linear
+        position.isRemovedOnCompletion = false
+        position.repeatCount = Float.greatestFiniteMagnitude
+        position.beginTime = CFTimeInterval(Double.random(in: 1 ..< 1_000).truncatingRemainder(dividingBy: Double(25)) / Double(100))
```

Comment:
I think give a specific type is better. No need to care about how compiler do the transfering.
My purpose is very clear, It reduces mental burden.

Reply:
<img width="681" alt="Screen Shot 2022-02-23 at 4 18 36 PM" src="https://user-images.githubusercontent.com/6863518/155282813-2a4f816c-8531-4e84-9cfc-d7315ebcfd50.png">
Missing a type will lead another result. I'm  reluctant to take this risk.

Reply:
25 and 100 is `Double` in `Double.random(in: 1 ..< 1_000).truncatingRemainder(dividingBy: 25) / 100`, or use 
`Double.random(in: 1 ..< 1_000).truncatingRemainder(dividingBy: 25.0) / 100.0`.

Reply:
2/200 return a Int value which is 0, But I give my purpose to get a Double.
My case is not the same with the code we talk about, I mean to explain that purposeful code lead a certain result, brief code lead a unknown result in some case.
Double(25) and 25.0 is the same.

Reply:
I mean, in case above, the compiler know 25 is `Double`, you can save `Double(25)` with `25.0` and the code is simpler.
`Double(25)` and `25.0` are the same value, but `25.0` is more swifty.
If you prefer to use `Double(25)`, it is ok.